packages/back-end/src/controllers/experiments.ts

```diff
@@ -1641,49 +1603,49 @@ export async function addScreenshot(
   const { url, description } = req.body;
   const changes: Changeset = {};
 
-  const exp = await getExperimentById(org.id, id);
+  const experiment = await getExperimentById(org.id, id);
 
-  if (!exp) {
+  if (!experiment) {
     res.status(403).json({
       status: 404,
       message: "Experiment not found",
     });
     return;
   }
 
-  if (exp.organization !== org.id) {
+  if (experiment.organization !== org.id) {
     res.status(403).json({
       status: 403,
       message: "You do not have access to this experiment",
     });
     return;
   }
-  req.checkPermissions("createAnalyses", exp.project);
+  req.checkPermissions("createAnalyses", experiment.project);
 
-  if (!exp.variations[variation]) {
+  if (!experiment.variations[variation]) {
     res.status(404).json({
       status: 404,
       message: "Unknown variation " + variation,
     });
     return;
   }
 
-  exp.variations[variation].screenshots =
-    exp.variations[variation].screenshots || [];
-  exp.variations[variation].screenshots.push({
+  experiment.variations[variation].screenshots =
+    experiment.variations[variation].screenshots || [];
+  experiment.variations[variation].screenshots.push({
```

Comment:
This is mutating the `experiment` variable.  Instead, you want to leave the original `experiment` alone and only modify the `changes` variable.